File: new-resource-detector/src/main/java/nl/haploid/resource/detector/AppConfiguration.java
package nl.haploid.resource.detector;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
import org.springframework.scheduling.annotation.EnableScheduling;

@Configuration
@EnableScheduling
public class AppConfiguration {

    @Bean
    public static PropertySourcesPlaceholderConfigurer propertyPlaceholderConfigurer() {
        return new PropertySourcesPlaceholderConfigurer();
    }
}

File: dirty-resource-detector/src/main/scala/nl/haploid/octowight/channel/dirty/configuration/AppConfiguration.scala
package nl.haploid.octowight.channel.dirty.configuration

import nl.haploid.octowight.JsonMapper
import org.springframework.context.annotation.{Bean, Configuration, PropertySource}
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
import org.springframework.scheduling.annotation.EnableScheduling

object AppConfiguration {

  @Bean def propertyPlaceholderConfigurer = new PropertySourcesPlaceholderConfigurer
}

@Configuration
@EnableScheduling
@PropertySource(value = Array("file:./local.properties"), ignoreResourceNotFound = true)
class AppConfiguration {

  @Bean def jsonMapper = new JsonMapper
}
